src/web/src/browser/presenter/actions/schematron.test.ts

```diff
@@ -0,0 +1,161 @@
+import { createPresenterMock, Presenter } from '..';
+
+describe('schematron', () => {
+  let presenter: Presenter;
+
+  beforeEach(() => {
+    presenter = createPresenterMock();
+    presenter.actions.schematron.setAssertions(MOCK_SCHEMATRON_ASSERTIONS);
+    presenter.actions.validator.setValidationReport({
+      validationReport: MOCK_VALIDATION_REPORT,
+      xmlText: '<xml></xml>',
+    });
+  });
+
+  describe('filtering', () => {
+    it('by role works', () => {
+      expect(presenter.state.schematron.filter).toEqual({
+        role: 'all',
+        text: '',
+      });
+      expect(presenter.state.schematron.roles).toEqual([
+        'all',
+        'error',
+        'info',
+        'warn',
+      ]);
+      presenter.actions.schematron.setFilterRole('error');
+      expect(presenter.state.schematron.filter).toEqual({
+        role: 'error',
+        text: '',
+      });
+      expect(presenter.state.schematron.schematronReport).toEqual({
+        groups: [
+          {
+            checks: {
+              checks: [
+                {
+                  fired: [],
+                  icon: {
+                    color: 'blue',
+                    sprite: 'help',
+                  },
+                  id: 'incorrect-role-association',
+                  isReport: false,
+                  message: 'incorrect role assertion message',
+                  role: 'error',
+                },
+              ],
+              summary: '1 checks',
+              summaryColor: 'green',
+            },
+            title: 'System Security Plan',
+          },
```

Comment:
B: can we either 1) discuss this the icon alignment with @sknep in the coming days as part of this PR process or 2) agree to open up an issue, perhaps prioritized for next sprint, to work short-term fixes on the icon alignment? Thanks!

<img width="480" alt="Screen Shot 2021-07-27 at 6 15 41 PM" src="https://user-images.githubusercontent.com/61464190/127234555-7ab8185f-b98f-4a32-b237-ef5ba5b90a27.png">

Reply:
Will look for another appropriate icon.